### PR TITLE
TIP-725 - Refactoring all index screen containing a basic grid ( Import/E…

### PR DESCRIPTION
**Pull request by Alban44**
…xport, Variant groups, Groups...)

[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**
Generalization of the refactoring made in the TIP-724 (#6025): each index screen containing a grid has been refactored with the new PEF architecture.
Each time, a new extension (i.e a new yaml file) is created and the corresponding twig file is cleaned up by removing all twig macro and by instantiating the right extension in Js instead.

This have been done for the following features:
* Import profiles
* Import history
* Variant groups
* Groups
* Export profiles
* Export history
* Families
* Group types
* Currencies
* Locales
* Channels
* Users
* User roles
* User groups
* Process tracker

**Important notes:**
 * In order to avoid sending the list of _groupTypes_ by the front-end to the back-end when displaying  group/variant group grids, the new key "default_query_params" in the yaml configuration file of datagrids has been created and used.
 * The new extension 'pim/job/export/edit/edit-button' has been created and used instead of generic use of 'pim/common/redirect' in job_instance (for being able to manage rights in EE)
 * The new extension 'pim/form/common/index/confirm-button' has been created in order to replace the twig macro 'confirmLink' 

_Remarks:_
 * Attributes index will be managed in another PR (because the creation popup is not a standard one)
 * An EE PR has been created for managing Proposals and Rules indexes

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/grid.js [hunk 1]
```diff
@@ -60,20 +65,30 @@ define([
...
+                this.urlParams.alias = alias;
+                this.urlParams.params = $.extend(true, {}, params);
+                this.urlParams[alias] = $.extend(true, {}, params);
+
+                var state = DatagridState.get(alias, ['view', 'filters', 'columns']) || {};
+                this.applyView(state.view, alias);
+                this.applyFilters(state.filters, alias);
+                this.applyColumns(state.columns, alias);
+
+                //TODO Manage columns for product form (when refactoring product form index)
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/grid.js]: Will those todo be fixed before the merge?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/job/export/edit/edit-button.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+/**
+ * Redirect button for edit job (needed for being able to manage rights in EE)
+ *
+ * @author    Alban Alnot <alban.alnot@consertotech.pro>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+define(
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/job/export/edit/edit-button.js]: I do'nt understand why it's needed

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/form/redirect.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<button class="AknButton AknButton--action AknButton--withIcon AknButtonList-item">
+<button class="AknButton AknButton--<%- actionType %> AknButton--withIcon AknButtonList-item">
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/form/redirect.html]: Not a huge fan of this class completion.
I prefer you have a "stupid" associative array in the JS for the class, like
```{
  action: 'AknButton--action',
  success: 'AknButton--success',
  ...
}```
Then pass the `buttonClass` to the template.

Finally, when you search for the occurences of "AknButton--success" in the code, you fill find it. (not with this case)

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/grid.js (hunk 1 above)
Comment (Alban44) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/grid.js]: For me, those TODO have to be fixed in two other appropriate PR as they are concerning product edit form and categories index that are still working with the old architecture.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/job/export/edit/edit-button.js (hunk 2 above)
Comment (Alban44) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/job/export/edit/edit-button.js]: In fact, we need it for being able to manage rights in the EE but only for job/export redirect buttons. By creating a dedicated buttons only for job/export, we can easily manage rights (by overriding isVisible method for instance) for only all job/export buttons.
May be a better way to do it is existing, if so, please enlighten me.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/grid.js (hunk 1 above)
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/grid.js]: ok

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/job/export/edit/edit-button.js (hunk 2 above)
Comment (Alban44) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/job/export/edit/edit-button.js]: Ok, after discussing, my mistake, it's useless: I will remove it!

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/form/redirect.html (hunk 3 above)
Comment (Alban44) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/form/redirect.html]: Ok, done!

**Pull request merged**